applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -438,7 +443,11 @@ def _create_builder_and_solver(self):
         linear_solver = self.get_linear_solver()
         if self.settings["block_builder"].GetBool():
             bs_params = self.settings["builder_and_solver_settings"]
-            builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(linear_solver, bs_params)
+            consider_lagrange_multiplier_constraint_resolution = bs_params["advanced_settings"]["consider_lagrange_multiplier_constraint_resolution"].GetString()
+            if consider_lagrange_multiplier_constraint_resolution == "none":
+                builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolver(linear_solver, bs_params)
+            else:
+                builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolverWithLagrangeMultiplier(linear_solver, bs_params)
```

Comment:
the `bs_params` and directly the `advanced_settings` and without any manipulation